### PR TITLE
[Improvement] Support particle types in particle trajectories

### PR DESCRIPTION
**Pull request by cphyc**
## PR Summary

This adds the possibility to select a specific particle type when doing particle trajectories

## PR Checklist

- [x] Code passes flake8 checker
- [x] New features are documented, with docstrings and narrative docs
- [x] Adds a test for any bugs fixed. Adds tests for new features.

### CODE REVIEW

File: yt/data_objects/particle_trajectories.py [hunk 1]
```diff
@@ -137,10 +139,18 @@ def __init__(self, outputs, indices, fields=None, suppress_logging=False):
...
     def keys(self):
         return self.field_data.keys()
 
+    def determine_field(self, field):
+        ds_first = self.data_series[0]
+        dd_first = ds_first.all_data()
+        if self.ptype:
+            return dd_first._determine_fields((self.ptype, field))
+        else:
+            return dd_first._determine_fields(field)
```
Comment (ngoldbaum) [inline yt/data_objects/particle_trajectories.py]: should probably use ('all', field), I think.

File: yt/data_objects/particle_trajectories.py [hunk 2]
```diff
@@ -137,10 +139,18 @@ def __init__(self, outputs, indices, fields=None, suppress_logging=False):
 
     def has_key(self, key):
         return (key in self.field_data)
-    
+
     def keys(self):
         return self.field_data.keys()
 
+    def determine_field(self, field):
```
Comment (ngoldbaum) [inline yt/data_objects/particle_trajectories.py]: maybe give this a different name that's more distinct from `_determine_fields`? If I wasn't reading closely I might miss the distinction.

File: yt/testing.py [hunk 3]
```diff
@@ -286,7 +286,7 @@ def fake_particle_ds(
             offsets.append(0.5)
         else:
             offsets.append(0.0)
-    data = {}
+    data = data if data else {}
```
Comment (ngoldbaum) [inline yt/testing.py]: good catch!

File: yt/data_objects/particle_trajectories.py [hunk 4]
```diff
@@ -221,20 +229,20 @@ def _get_data(self, fields):
         fds = {}
         new_particle_fields = []
         for field in missing_fields:
-            fds[field] = dd_first._determine_fields(field)[0]
+            fds[field] = self.get_full_field_name(field)[0]
```
Comment (ngoldbaum) [inline yt/data_objects/particle_trajectories.py]: This field isn't going to necessarily be a particle field, so I think using `_determine_fields` here is correct. I bet that will fix the test failure.

Comment (cphyc) [reply yt/data_objects/particle_trajectories.py]: I corrected it.

File: yt/data_objects/particle_trajectories.py [hunk 5]
```diff
@@ -137,10 +139,16 @@ def __init__(self, outputs, indices, fields=None, suppress_logging=False):
 
     def has_key(self, key):
         return (key in self.field_data)
-    
+
     def keys(self):
         return self.field_data.keys()
 
+    def get_full_field_name(self, field):
```
Comment (jzuhone) [inline yt/data_objects/particle_trajectories.py]: Very minor nit--this seems like it should be `_get_full_field_name`.

File: yt/data_objects/time_series.py [hunk 6]
```diff
@@ -418,6 +418,8 @@ def particle_trajectories(self, indices, fields=None, suppress_logging=False):
         suppress_logging : boolean
             Suppress yt's logging when iterating over the simulation time
             series. Default: False
+        ptype : str, optional
+            Only use this particle type
```
Comment (jzuhone) [inline yt/data_objects/time_series.py]: We should say something like `Default: None, which uses all particle types`. Also, can you put this element of the docstring in the `ParticleTrajectories` class itself?

File: yt/data_objects/particle_trajectories.py [hunk 7]
```diff
@@ -137,10 +139,16 @@ def __init__(self, outputs, indices, fields=None, suppress_logging=False):
...
+        ds_first = self.data_series[0]
+        dd_first = ds_first.all_data()
+        ptype = self.ptype if self.ptype else 'all'
+        return dd_first._determine_fields((ptype, field))
```
Comment (jzuhone) [inline yt/data_objects/particle_trajectories.py]: Do we have to do `_determine_fields` on an object or can we just do something like `ds._get_field_info`? @ngoldbaum ?

Comment (ngoldbaum) [reply yt/data_objects/particle_trajectories.py]: Either will work and give the same answer.

Comment (ngoldbaum) [reply yt/data_objects/particle_trajectories.py]: This is fine as-is I think.

**Comment by ngoldbaum** [comment]
@yt-fido test this please

**Pull request merged**